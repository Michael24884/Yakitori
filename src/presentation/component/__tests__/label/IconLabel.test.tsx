/**
 * @format
 */
import React from 'react';
import {Text, Image} from 'react-native';

import {IconLabel} from '@components';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('render correctly', () => {
  const instance = renderer.create(<IconLabel />);
  expect(instance.toJSON()).toMatchSnapshot();
});
it('render prefix', () => {
  const instance = renderer.create(<IconLabel text="test" />);
  const textView = instance.root.findByType(Text);
  expect(textView.props.children).toBe('test');
  expect(instance.root.findAllByType(Image)).toHaveLength(0);

  // test render Prefix
  instance.update(<IconLabel prefix={<Text children="test" />} />);
  expect(instance.root.findByProps({children: 'test'})).toBeDefined();
  expect(instance.root.findAllByType(Image)).toHaveLength(0);

  // test render Prefix
  const prefixIcon = 1;
  instance.update(<IconLabel prefixIcon={prefixIcon} />);
  expect(instance.root.findByType(Image)?.props?.source).toEqual(prefixIcon);
});
