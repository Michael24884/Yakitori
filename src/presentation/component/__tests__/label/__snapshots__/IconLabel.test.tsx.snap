// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "marginHorizontal": 8,
        },
        undefined,
      ]
    }
  />
</View>
`;
